tools/teachablelearnsethelper/teachable_learnset_helper.py

```diff
@@ -0,0 +1,89 @@
+import glob
+import re
+import json
+
+def parse_mon_name(name):
+    return re.sub('(?!^)([A-Z]+)', r'_\1', name).upper()
+    
+tm_moves = []
+tutor_moves = []
+
+# scan incs
+incs_to_check =  glob.glob('./data/scripts/*.inc') # all .incs in the script folder
+incs_to_check += glob.glob('./data/maps/*/scripts.inc') # all map scripts
+for file in incs_to_check:
+    with open(file, 'r') as f2:
+        raw = f2.read()
+    if 'special ChooseMonForMoveTutor' in raw:
+        for x in re.findall('setvar VAR_0x8005, (MOVE_.*)', raw):
+            if not x in tutor_moves:
+                tutor_moves.append(x)
```

Comment:
I think there's a few ways in which is this a little weird:
1. Somebody might use `setvar VAR_0x8005, MOVE_...` for something other than `ChooseMonForMoveTutor`, e.g. they might have a special that teaches a particular move (which is in `VAR_0x8005`, a sensible choice because `VAR_0x8004` contains the result of `ChoosePartyMon`)..
2. If somebody did, e.g. `copyvar VAR_0x8005, VAR_MOVE_TO_TEACH` then it won't be detected (and they'd probably be doing that in a way that makes it difficult to know what `VAR_MOVE_TO_TEACH` contains, at the very least through a `call`, but it could be a variable set in an entirely different script). Related, (and somewhat more likely) somebody could write a special which writes the move ID into `gSpecialVars_0x8005`.
3. If you write `@ setvar VAR_0x8005, MOVE_TACKLE` I think that would be picked up. This is both good and bad, bad because commenting things out should work, but good because it gives a way to explicitly list teachable moves in cases where it can't be picked up automatically (like 2).
4. Very, very minor, but this will find scripts that aren't OW scripts in `data/scripts` (i.e. not referenced by `data/event_scripts.s`). This is very unlikely to matter unless people write other script languages that can also reference OW vars.
5. If somebody happened to write move tutors in a macro they won't be detected. (I personally think a macro is a bad idea because I want to see what's happening to the special vars, but I could see other people thinking differently and could easily see a change like that making it into here or pret\*).

I don't have a workable suggestion for these because I think you'd have to do static analysis to work out which value is in `VAR_0x8005` when `special ChooseMonForMoveTutor` runs. In practice it probably won't come up except for people who are really pushing the boundaries of the scripting language. afaict MoT wouldn't see any bugs, although we do hit 1-3 points above and almost 4 (we just happen not to ever assign a move to `VAR_0x8005`).

tl;dr not using a parser, but I don't expect it to cause a problem. It's worth noting that something similar to 2 exists in vanilla: [`GetBattleFrontierTutorMoveIndex`](<https://github.com/pret/pokeemerald/blob/6c96641035099db8a2c1e680fb8793f1b29cf2fd/src/field_specials.c#L3188>) (it generates an index rather than a move ID because vanilla's tutors take indices not move IDs).

btw, I'd like to be really explicit that I love the idea of determining the smallest possible data structures at build time.

\* Not pret for specifically move tutors, because they work differently there.